Sources/ZIPFoundation/Archive+Writing.swift

```diff
@@ -118,7 +118,7 @@ extension Archive {
         let centralDirectory = try self.writeCentralDirectoryStructure(localFileHeader: localFileHeader,
                                                                        relativeOffset: offset,
                                                                        externalFileAttributes: externalFileAttributes)
-        if startOfCentralDirectory > UINT32_MAX {
```

Comment:
This change would violate the ZIP specification. 
`Int.max` is platform dependent. It's 2^63 − 1 on arm64 and x86_64 and 2^31 - 1 on older ARM versions (for example iPhone 5 & 5c) and i386.  
ZIP (without the ZIP64 extension) restricts the start of the central directory to be within UInt32 range (2^32 - 1).  
So a better approach would be to use `Int(UInt32.max)` here.